application/src/Controller/Admin/ResourceTemplateController.php

```diff
@@ -40,6 +50,271 @@ public function showDetailsAction()
+     * @return array
+     */
+    protected function flagValid(array $import)
+    {
+        $vocabs = [];
+        $dataTypes = ['literal', 'uri', 'resource'];
+
+        $getVocab = function ($namespaceUri) use (&$vocabs) {
+            if (isset($vocabs[$namespaceUri])) {
+                return $vocabs[$namespaceUri];
+            }
+            $vocab = $this->api()->searchOne('vocabularies', [
+                'namespace_uri' => $namespaceUri,
+            ])->getContent();
+            if ($vocab) {
+                $vocabs[$namespaceUri] = $vocab;
+                return $vocab;
+            }
+            return false;
+        };
+
+        if ($vocab = $getVocab($import['o:resource_class']['vocabulary_namespace_uri'])) {
+            $import['o:resource_class']['vocabulary_prefix'] = $vocab->prefix();
+            $class = $this->api()->searchOne('resource_classes', [
+                'vocabulary_namespace_uri' => $import['o:resource_class']['vocabulary_namespace_uri'],
+                'local_name' => $import['o:resource_class']['local_name'],
+            ])->getContent();
+            if ($class) {
+                $import['o:resource_class']['o:id'] = $class->id();
+            }
+        }
+
+        foreach ($import['o:resource_template_property'] as $key => $property) {
+            if ($vocab = $getVocab($property['vocabulary_namespace_uri'])) {
+                $import['o:resource_template_property'][$key]['vocabulary_prefix'] = $vocab->prefix();
+                $prop = $this->api()->searchOne('properties', [
+                    'vocabulary_namespace_uri' => $property['vocabulary_namespace_uri'],
+                    'local_name' => $property['local_name'],
+                ])->getContent();
+                if ($prop) {
+                    $import['o:resource_template_property'][$key]['o:property'] = ['o:id' => $prop->id()];
+                    if (in_array($import['o:resource_template_property'][$key]['data_type_name'], $dataTypes)) {
+                        $import['o:resource_template_property'][$key]['o:data_type'] = $import['o:resource_template_property'][$key]['data_type_name'];
+                    }
+                }
+            }
+        }
+
+        return $import;
+    }
+
+    /**
+     * Verify that the import format is valid.
+     *
+     * @param array $import
+     * @return bool
+     */
+    protected function importIsValid($import)
+    {
+        if (!is_array($import)) {
+            // invalid format
+            return false;
+        }
+
+        if (!isset($import['o:label']) || !is_string($import['o:label'])) {
+            // missing or invalid label
+            return false;
+        }
+
+        // Validate class.
+        if (!isset($import['o:resource_class']) || !is_array($import['o:resource_class'])) {
+            // missing or invalid o:resource_class
+            return false;
+        }
+        if (!array_key_exists('vocabulary_namespace_uri', $import['o:resource_class'])
```

Comment:
Does this prevent import of templates that don't have an assigned resource class?